src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java

```diff
@@ -0,0 +1,119 @@
+        this(parametersList, FileBuildParameterFactory.NoFilesFoundEnum.SKIP);
+    }
+
+    public FileBuildParameterFactory.NoFilesFoundEnum getNoFilesFoundAction() {
+        return noFilesFoundAction;
+    }
+
+    @Override
+    public List<AbstractBuildParameters> getParameters(AbstractBuild<?, ?> build, TaskListener listener) throws IOException, InterruptedException, AbstractBuildParameters.DontTriggerException {
+        List<AbstractBuildParameters> result = Lists.newArrayList();
+        int totalFiles = 0;
+        for (final Tuple t : parametersList) {
+            // save them into the master because FileParameterValue might need files after the slave workspace have disappeared/reused
+            FilePath target = new FilePath(build.getRootDir()).child("parameter-files");
+                int k = build.getWorkspace().copyRecursiveTo(t.filePattern, target);
+                totalFiles += k;
+                if (k > 0) {
+                    for (final FilePath f : target.list(t.filePattern)) {
+                        LOGGER.fine("Triggering build with " + f.getName());
+
+                        result.add(new AbstractBuildParameters() {
+                            @Override
+                            public Action getAction(AbstractBuild<?, ?> build, TaskListener listener) throws IOException, InterruptedException, DontTriggerException {
+                                assert f.getChannel() == null;    // we copied files locally. This file must be local to the master
+                                FileParameterValue fv = new FileParameterValue(t.parameterName, new File(f.getRemote()), f.getName());
+                                if ($setLocation != null) {
+                                    try {
+                                        $setLocation.invoke(fv, t.parameterName);
+                                    } catch (IllegalAccessException e) {
+                                        // be defensive as the core might change
+                                    } catch (InvocationTargetException e) {
+                                        // be defensive as the core might change
+                                    }
+                                }
+                                return new ParametersAction(fv);
+                            }
+                        });
+                    }
+                }
+        }
+        if (totalFiles ==0) {
+            noFilesFoundAction.failCheck(listener);
+        }
+
+        return result;
+    }
+
+
+    @Extension
+    public static class DescriptorImpl extends AbstractBuildParameterFactoryDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Multiple Binary Files (not meant to be used)";
+        }
+    }
+
+    private static Method $setLocation;
+
+    static {
+        // work around NPE fixed in the core at 4a95cc6f9269108e607077dc9fd57f06e4c9af26
```

Comment:
That is in 1.518, so add a `TODO` comment explaining what to delete when our baseline goes past that.